src/pybind/mgr/dashboard_v2/components/rgw_client.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import re
+from awsauth import S3Auth
+from ..settings import Settings, Options
+from ..components.rest_client import RestClient, RequestException
+from ..tools import build_url, isset
+from .. import mgr, logger
+
+
+class RgwClient(RestClient):
+    _SYSTEM_USERID = None
+    _ADMIN_PATH = None
+    _host = None
+    _port = None
+    _ssl = None
+    _user_instances = {}
+
+    class NoCredentialsException(Exception):
+        def __init__(self):
+            super(RgwClient.NoCredentialsException,
+                  self).__init__('No RGW credentials found')
+
+    @staticmethod
+    def _determine_rgw_addr():
```

Comment:
doesn't seem to be a need for this to be a staticmethod for this class as it is only used once. This could very well be a function